src/NHibernate.Validator/Event/ValidatePreCollectionUpdateEventListener.cs

```diff
@@ -1,17 +1,25 @@
 using System;
+using System.Threading;
+using System.Threading.Tasks;
 using NHibernate.Event;
 
 namespace NHibernate.Validator.Event
 {
 	[Serializable]
 	public class ValidatePreCollectionUpdateEventListener : ValidateEventListener, IPreCollectionUpdateEventListener
 	{
+		public Task OnPreUpdateCollectionAsync(PreCollectionUpdateEvent @event, CancellationToken cancellationToken)
+		{
+			OnPreUpdateCollection(@event);
+			return Task.CompletedTask;
+		}
+
 		public void OnPreUpdateCollection(PreCollectionUpdateEvent @event)
 		{
 			var owner = @event.AffectedOwnerOrNull;
-			if(!ReferenceEquals(null,owner))
+			if (!ReferenceEquals(null, owner))
 			{
-				Validate(owner, @event.Session.EntityMode);
+				Validate(owner, EntityMode.Poco);

```

Comment:
Agree i have proposed dirty solution.
Seems this will be correct approach in this case:

Validate(owner, @event.Session.GetEntityPersister(@event.Session.GetEntityName(owner), owner).EntityMode);

Will fix